src/os_sensor_nodelet.cpp

```diff
@@ -287,12 +293,33 @@ class OusterSensor : public OusterClientBase {
             config_flags |= ouster::sensor::CONFIG_UDP_DEST_AUTO;
         }
 
+        if (sensor::in_multicast(udp_dest)) {
+            if (is_arg_set(mtp_dest_arg)) {
+                NODELET_INFO("Will recieve data via multicast on %s", mtp_dest_arg.c_str());
+                this->mtp_dest = mtp_dest_arg;
+            } else {                
+                NODELET_INFO("mtp_dest arg not defined, will recieve data via multicast"
+                             " on first available interface");
+                this->mtp_dest = std::string{};
+            }
+
+            this->mtp_main = mtp_main_arg;
+        }
+
         return std::make_pair(config, config_flags);
     }
 
     void configure_sensor(const std::string& hostname,
-                          const sensor::sensor_config& config,
+                          sensor::sensor_config& config,
                           int config_flags) {
+        if (sensor::in_multicast(config.udp_dest.value()) && !mtp_main) {
```

Comment:
This also fails when `config.udp_dest` is not set 
```suggestion
        if (config.udp_dest && sensor::in_multicast(config.udp_dest.value()) && !mtp_main) {
```